lesson16/index.js

```diff
@@ -0,0 +1,128 @@
+// const url = "https://myjson.dit.upm.es/api/bins/hyv9";
+const url = "https://myjson.dit.upm.es/api/bins/d9vx";
+const newsBlock = document.querySelector(".news");
+const newsTabs = document.querySelector(".news_Tabs");
+
+const createNewsTabs = (allData) => {
+  allData.forEach((content, index) => {
+    const li = document.createElement("li");
+    li.classList.add("tab_item");
+    li.id = content.category;
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+
+  const selectedData = allData.find((data) => data.isActive === true);
+  primaryImageWrapper.classList.add("news_Img");
+  const img = document.createElement("img");
+  img.src = selectedData.image.img;
+  primaryImageWrapper.appendChild(img);
+
+  renderNewsList(selectedData);
+  document.getElementById(selectedData.category).classList.add("is-active");
+
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+
+    // タブのindexを取得し、スタイル変更
```

Comment:
IMO
I feel uncomfortable defining tab events within createNewsContent.

How about separating it into functions and executing them with init...?
```js
const setTabClickEvent = (allData) => {
  // タブクリック時の処理
  newsTabs.addEventListener("click", (event) => {
    // event.targetにis-activeをつける
    // event.targetのidを取得し、それに紐づく記事と画像を取得
    const target = event.target;

    // タブのindexを取得し、スタイル変更
    const activeTab = newsTabs.querySelector(".is-active");
    activeTab.classList.remove("is-active");
    target.classList.add("is-active");

    const targetData = allData.find((d) => d.category === target.id);
    setCategoryImage(targetData);
    renderNewsList(targetData);
  });
}

```